Packs/GenericSQL/Integrations/GenericSQL/GenericSQL.py

```diff
@@ -165,8 +167,16 @@ def _create_engine_and_connect(self) -> sqlalchemy.engine.base.Connection:
             cache_key = self._get_cache_string(str(db_url), ssl_connection)
             engine = cache.get(cache_key, None)
             if engine is None:  # (first time or expired) need to initialize
+
+                engines = getattr(sqlalchemy, ENGINES, {})
+
+                if cache_key in engines:
+                    engines[cache_key].dispose()
```

Comment:
How does SQLA behave if we call `.dispose` _after_ the connection is closed (say, it's been a day since the last connection)?

Reply:
It's fine, since the .dispose() method is an engine's method, you are actually disposing of the engine's object, which contains the connections.